app/browse-solutions/ui.test.js

```diff
@@ -0,0 +1,122 @@
+  .afterEach(async (t) => {
+    const isDone = nock.isDone();
+    if (!isDone) {
+      nock.cleanAll();
+    }
+
+    await t.expect(isDone).ok('Not all nock interceptors were used!');
+  });
+
+test('should navigate to home page when click Go back', async (t) => {
+  await pageSetup(t);
+
+  const getLocation = ClientFunction(() => document.location.href);
+
+  const goBackLink = Selector('[data-test-id="go-back-link"] a');
+
+  await t
+    .expect(goBackLink.exists).ok()
+    .click(goBackLink)
+    .expect(getLocation()).eql('http://localhost:1234/');
+});
+
+test('should render Browse Foundation Solutions container', async (t) => {
+  await pageSetup(t);
+
+  const foundationSolutions = Selector('[data-test-id="foundation-solutions-promo"]');
+
+  await t
+    .expect(foundationSolutions.exists).ok()
+    .expect(foundationSolutions.find('h3').innerText).eql(content.foundationPromoHeading)
+    .expect(foundationSolutions.find('p').innerText).eql(content.foundationPromoDescription);
+});
+
+test('should render Browse All Solutions container', async (t) => {
+  await pageSetup(t);
+
+  const allSolutions = Selector('[data-test-id="all-solutions-promo"]');
+
+  await t
+    .expect(allSolutions.exists).ok()
+    .expect(allSolutions.find('h3').innerText).eql(content.allPromoHeading)
+    .expect(allSolutions.find('p').innerText).eql(content.allPromoDescription);
+});
+
+test('should navigate to browse all solutions page', async (t) => {
+  nock('http://localhost:8080')
+    .get('/api/v1/Solutions')
+    .reply(200, aSolutionList);
+
+  await pageSetup(t);
+
+  const getLocation = ClientFunction(() => document.location.href);
+
+  const browseAllSolutionsLink = Selector('[data-test-id="all-solutions-promo"]').find('a');
+
+  await t
+    .expect(browseAllSolutionsLink.exists).ok()
+    .click(browseAllSolutionsLink)
+    .expect(getLocation()).contains('/all');
+});
+
+test('should navigate to browse foundation solutions page', async (t) => {
+  nock('http://localhost:8080')
+    .get('/api/v1/Solutions/Foundation')
+    .reply(200, aFoundationSolutionList);
+
+  await pageSetup(t);
+
+  const getLocation = ClientFunction(() => document.location.href);
+
+  const foundationSolutionsLink = Selector('[data-test-id="foundation-solutions-promo"]').find('a');
+
+  await t
+    .expect(foundationSolutionsLink.exists).ok()
+    .click(foundationSolutionsLink)
+    .expect(getLocation()).contains('/foundation');
+});
+
+test('should render the error page after navigating to browse all solutions page when receiving an error from the solutions api endpoint', async (t) => {
+  nock('http://localhost:8080')
+    .get('/api/v1/Solutions')
+    .reply(500, {});
+
+  await pageSetup(t);
+
+  const browseAllSolutionsLink = Selector('[data-test-id="all-solutions-promo"]').find('a');
+
+  await t.click(browseAllSolutionsLink);
+
+  const errorTitle = Selector('[data-test-id="error-page-title"]');
+
+  await t
+    .expect(errorTitle.exists).ok();
+});

```

Comment:
I think this test sits better in the solution-list page rather than here. As far as browse-solution there is nothing in here that calls the API that would cause an error.